server/config/pooling.go

```diff
@@ -1,41 +1,38 @@
 package config
 
-import "time"
+import (
+	"time"
+)
 
-func (c *Config) PoolingConfig() PoolingConfig {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+var poolingOptions = options{
+	{"poolingConfig.maxWaitTimeForTrace", "30s", "pooling config: max wait time for trace"},
+	{"poolingConfig.retryDelay", "1s", "pooling config: interval between poll retries"},
+}
 
-	return c.config.PoolingConfig
+func init() {
+	configOptions = append(configOptions, poolingOptions...)
 }
 
-func (c *Config) SetPoolingConfig(pc PoolingConfig) {
+func (c *Config) PoolingMaxWaitTimeForTraceDuration() time.Duration {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	c.config.PoolingConfig = pc
+	maxWaitTimeForTrace := c.vp.GetDuration("poolingConfig.maxWaitTimeForTrace")
+	if maxWaitTimeForTrace == 0 {
```

Comment:
Can we print a Warning log here saying that this value is malformed and we are assuming a default value? It could be good to avoid implicit behaviors for our users.